pkg/virtualgarden/etcd.go

```diff
@@ -36,7 +36,22 @@ const (
 // DeployETCD generates a dedicated CA, two server certificates (for main and events etcd), and one client
 // certificate. It deploys them as secrets together with the bootstrap config map. Finally, it creates a service for
 // etcd that can be used by clients and deploys it as a stateful set with a persistent volume for its data.
+// It waits until the stateful sets are ready.
 func (o *operation) DeployETCD(ctx context.Context) error {
+	if err := o.deployETCDResources(ctx); err != nil {
+		return err
+	}
+
+	for _, role := range []string{ETCDRoleMain, ETCDRoleEvents} {
+		if err := waitForStatefulSetReady(ctx, o.client, emptyETCDStatefulSet(o.namespace, role)); err != nil {
+			return err
+		}
+	}
```

Comment:
The deployment of the main and the event etcd should be put in two separate goroutines (another flow step).
Waiting for the etcd statefulset to become ready can take some time (e.g Volume creation).
Flow
 - deployETCDResourcesMain -> wait for statefulset(main)
 -  deployETCDResourcesEvents -> wait for statefulset(events)

Reply:
We currently deploy the main and event etcd first. After the deployment of both has finished, we wait for them to become ready. If waiting of the stateful sets is the dominating factor, we do not save much time if we parallelise the deployment.

Reply:
Yes, right. So we get the most gain from a more concurrent design of the `waitForStatefulSetReady `. 
I ll leave that up to you if you either want to only parallelize the `waitForStatefulSetReady ` via goroutines directly, or take advantage of the `flow` that does it for you.

For the latter, I suggested splitting the deployment & waiting for statefulset to become ready into two dedicated flows (one for each etcd-type).

Reply:
Also when we parallelise the waitForStatefulSetReady call, we will not save much time because if we wait for the first stateful set to become ready, also the other one is making progress in the meantime and then the second call of waitForStatefulSetReady is usually much faster. For the moment we would prefer to let the implementation as it is and concentrate on the open functionality. We still need to implement the etcd backup for aws and alicloud.

Reply:
okay sure, won't press it.